addon/webextension/catcher.js

```diff
@@ -0,0 +1,74 @@
+window.catcher = (function () {
+  let exports = {};
+
+  let handler;
+
+  let queue = [];
+
+  exports.unhandled = function (error, info) {
+    console.error("Unhandled error:", error, info);
+    let e = makeError(error, info);
+    if (! handler) {
+      queue.push(e);
+    } else {
+      handler(e);
+    }
+  };
+
+  /** Turn an exception into an error object */
+  function makeError(exc, info) {
+    let result;
+    if (exc.fromMakeError) {
+      result = exc;
+    } else {
+      result = {
+        fromMakeError: true,
+        name: exc.name || "ERROR",
+        message: exc+"",
+        stack: exc.stack
+      };
+      for (let attr in exc) {
+        result[attr] = exc[attr];
```

Comment:
I'm relying on the serialization actually doing that filtering, since we'd have to do a deep inspection to really be sure.  As far as I know anything that causes a problem with serialization will simply be omitted?